dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -1412,13 +1412,13 @@ private void addSet(final String assetKey, final AssetTypes assetTypes) {
 		 * The current thread wait until all the dependencies are processed
 		 * @throws ExecutionException
 		 */
-		private void waitForAll() throws ExecutionException {
+		private void startAndWait() throws ExecutionException {
 			final String submitterName = "DependencyManagerSubmitter" + Thread.currentThread().getName();
```

Comment:
why are you creating a new submitter per thread

Reply:
done https://github.com/dotCMS/core/pull/20173/commits/e491040a5cc382fa8a25457ba027679b4afa86b7#diff-abb9df68dc16a9532f21f0164ea5c958a1f3462cce8aa09bcce2dd984da40910R1444